thumberd/thumberd.go

```diff
@@ -528,31 +536,32 @@ func thumbServer(w http.ResponseWriter, r *http.Request, sem chan int) {
 	atomic.AddInt64(&http_stats.ok, 1)
 }
 
-func fetchImageWithCorrectFormat(src io.Reader) (imageBlob []byte, err error) {
+func fetchImageWithCorrectFormat(src io.Reader) (imageBlob []byte, format int, err error) {
 	buf := make([]byte, 20)
 	_, err = io.ReadFull(src, buf)
 	if err != nil {
-		return nil, err
+		return nil, FORMAT_OTHER, err
 	}
 
 	// FORMAT_OTHER means this file format is not supported.
 	// For security purposes, we are restricting our input image format.
-	if detectImageFormat(buf) == FORMAT_OTHER {
+	format = detectImageFormat(buf)
+	if format == FORMAT_OTHER {
 		msg := "input image format is not supported"
```

Comment:
Can we change the error message to:
"Invalid data retrieved"

Reply:
Sure, updated.